# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositorBridgeParent.cpp
# Commit: ae871e76cfae
# Full Hash: ae871e76cfae261ee2a43f039dabb7f1a4d8c4a8
# Author: Botond Ballo <botond@mozilla.com>
# Date: 2019-08-22 21:54:53
# Description:
#   Bug 1575498 - Null-check state->mParent in GetStateForRoot(). r=hsivonen
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D42909
# ==============================================================================

diff -r bf99d92d7805 -r ae871e76cfae gfx/layers/ipc/CompositorBridgeParent.cpp
--- a/gfx/layers/ipc/CompositorBridgeParent.cpp	Thu Aug 22 17:08:22 2019 +0300
+++ b/gfx/layers/ipc/CompositorBridgeParent.cpp	Thu Aug 22 07:15:52 2019 +0000
@@ -2259,7 +2259,7 @@
   // the mApzcTreeManagerParent from that. This should also work with nested
   // content processes, because RootLayerTreeId() will bypass any intermediate
   // processes' ids and go straight to the root.
-  if (state) {
+  if (state && state->mParent) {
     LayersId rootLayersId = state->mParent->RootLayerTreeId();
     itr = sIndirectLayerTrees.find(rootLayersId);
     state = (sIndirectLayerTrees.end() != itr) ? &itr->second : nullptr;
